src/R/Editor/Impl/Completion/Providers/PackageFunctionCompletionProvider.cs

```diff
@@ -153,19 +156,23 @@ public class PackageFunctionCompletionProvider : IRCompletionListProvider, IRHel
         /// </summary>
         /// <param name="context"></param>
         /// <returns></returns>
-        private IEnumerable<IPackageInfo> GetAllFilePackages(RCompletionContext context) {
+        private async Task<IEnumerable<IPackageInfo>> GetAllFilePackagesAsync(RCompletionContext context) {
             _loadedPackagesProvider?.Initialize();
 
             IEnumerable<string> loadedPackages = _loadedPackagesProvider?.GetPackageNames() ?? Enumerable.Empty<string>();
             IEnumerable<string> filePackageNames = context.AstRoot.GetFilePackageNames();
             IEnumerable<string> allPackageNames = PackageIndex.PreloadedPackages.Union(filePackageNames).Union(loadedPackages);
 
-            return allPackageNames
-                .Select(packageName => GetPackageByName(packageName))
+            var list = new List<IPackageInfo>();
+            foreach (var name in allPackageNames) {
+                var info = await _packageIndex.GetPackageInfoAsync(name);
```

Comment:
Can we then at least fire off all the async requests at once, and `WaitAll` a combination of them, so that throughput is maximized? It seems like in the worst case, this will produce a latency that is a multiple of the underling network latency by the number of packages fetched - meaning seconds of time, which would be very noticeable.